### PR TITLE
Create ingest-from-redhat-amq-streams.md

### LINKED ISSUE #1996: Doc: MQTT source connector
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/15388

### Which part(s) of the docs might be affected or should be updated? And how?

Add doc for MQTT source connector

### Reference

_No response_

### PR DESCRIPTION
**Pull request by fahadullahshah261**
I'm submitting this PR regarding data ingestion from the RedHat AMQ Stream. Please review it. Thanks.

### CODE REVIEW

**Comment by hengm3467** [comment]
@fahadullahshah261 Please add an entry to the sidebar.js file that is in the root directory.

File: docs/ingest/ingest-from-redhat-amq-streams.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+---
+<head>
+  <link rel="canonical" href="https://docs.risingwave.com/docs/current/ingest-from-redhat-amq-streams/" />
+</head>
+
+You can ingest data from RedHat AMQ Streams into RisingWave by using the Kafka source connector in RisingWave.
+
+ [AMQ Streams](https://developers.redhat.com/products/amq/overview) offers a distributed backbone that allows microservices and other applications to share data with extremely high throughput and extremely low latency, it is based on the  [Strimzi](http://strimzi.io/) and [Apache Kafka](http://kafka.apache.org/) projects. It has the features of publish and subscribe, long-term data retention, advanced queueing, replayable events, and, partition messages for scalability.
+
+## Prerequisites[](https://docs.risingwave.com/docs/current/ingest-from-nats/#prerequisites)
```
Comment (neverchanje) [inline docs/ingest/ingest-from-redhat-amq-streams.md]: Why did you add this link here?

**Review by neverchanje — APPROVED** [review]
Rest LGTM

**Review by WanYixian — APPROVED** [review]
Edit a bit and LGTM now

**Review by tabVersion — APPROVED** [review]

**Comment by fahadullahshah261** [comment]
Hi, I am submitting this PR regarding data ingestion from the MQTT server. Please review it. Thanks.

**Comment by WanYixian** [comment]
> Hi, I am submitting this PR regarding data ingestion from the MQTT server. Please review it. Thanks.

Just add the sidebar for MQTT server and remove the extra links in the doc. LGTM now

**Comment by hengm3467** [comment]
@neverchanje @tabVersion @WanYixian Fahad added another topic to this PR. It's about the MQTT integration. Please take a look at that topic as well. Thanks!

File: docs/ingest/ingest-from-mqtt.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+For example, we create a topic named `iot_data` with the data from various IoT devices at a given timestamp, including temperature, humidity readings, and a status field indicating whether the device is in a normal or abnormal state.  For more information to learn about MQTT and get started with it, refer to the [MQTT guide](https://mqtt.org/getting-started/).
+
+## Ingest data into RisingWave
+
+We create a table in RisingWave to persist the data by using `CREATE TABLE` and specifying the connection settings and data format.
+
+### Syntax
+
+```sql
+CREATE TABLE IF NOT EXISTS table_name
```
Comment (neverchanje) [inline docs/ingest/ingest-from-mqtt.md]: I suggest creating a source rather than a table to clarify that persistence is not necessary for MQTT source.

File: docs/ingest/ingest-from-mqtt.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+(
+   column_name data_type PRIMARY KEY,
+   -- Add more columns if needed: column_name2 data_type2, column_name3 data_type3, ...
+
+   -- Define the primary key constraint if needed
+   -- PRIMARY KEY (column_name, ...)
+
+)
+WITH (
+   connector = 'nats',
```
Comment (neverchanje) [inline docs/ingest/ingest-from-mqtt.md]:
```suggestion
   connector = 'mqtt',
```

**Comment by neverchanje** [comment]
Would you mind adding a table for source options?
https://github.com/risingwavelabs/risingwave/blob/main/src/connector/with_options_source.yaml#L258 You can refer to this file for details.

File: docs/ingest/ingest-from-mqtt.md (hunk 2 above)
Comment (fahadullahshah261) [reply docs/ingest/ingest-from-mqtt.md]: @neverchanje, I tried to create a source but, I think source implementation is different than the table which is why it might not be supported yet. I had the same observation when I was ingesting data from the Neon.

Comment (fahadullahshah261) [reply docs/ingest/ingest-from-mqtt.md]: > @neverchanje, I tried to create a source but, I think source implementation is different than the table which is why it might not be supported yet. I had the same observation when I was ingesting data from the Neon.

The source also works and will update the PR accordingly.

Comment (neverchanje) [reply docs/ingest/ingest-from-mqtt.md]: Thanks! If source doesn't work, it would mean that this connector is not well-supported, then we should not document it.

File: docs/ingest/ingest-from-mqtt.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+---
+id: ingest-from-mqtt
+title: Ingest data from MQTT server
```
Comment (hengm3467) [inline docs/ingest/ingest-from-mqtt.md]:
```suggestion
title: Ingest data from MQTT brokers
```

File: docs/ingest/ingest-from-mqtt.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+description: Ingest data from MQTT server into RisingWave.
```
Comment (hengm3467) [inline docs/ingest/ingest-from-mqtt.md]:
```suggestion
description: Ingest data from MQTT brokers into RisingWave.
```

File: docs/ingest/ingest-from-mqtt.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+slug: /ingest-from-mqtt
+---
+<head>
+  <link rel="canonical" href="https://docs.risingwave.com/docs/current/ingest-from-mqtt/" />
+</head>
+
+You can ingest data from the MQTT server into RisingWave by using the MQTT source connector in RisingWave.
```
Comment (hengm3467) [inline docs/ingest/ingest-from-mqtt.md]:
```suggestion
You can ingest data from a MQTT broker into RisingWave by using the MQTT source connector in RisingWave.
```

File: docs/ingest/ingest-from-mqtt.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+The [Message Queuing Telemetry Transport](https://mqtt.org/) (MQTT) protocol is ideal for connecting remote devices with a small code footprint and minimal network bandwidth. MQTT today is used in a wide variety of industries, such as automotive, manufacturing, telecommunications, oil and gas, etc.
+
+## Prerequisites
+
+Before ingesting data from MQTT into RisingWave, please ensure the following:
+
+- The MQTT server is running and accessible from your RisingWave cluster.
```
Comment (hengm3467) [inline docs/ingest/ingest-from-mqtt.md:19]: Let's use "MQTT broker" instead of "MQTT server" in this topic.

File: docs/ingest/ingest-from-redhat-amq-streams.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+You can ingest data from RedHat AMQ Streams into RisingWave by using the Kafka source connector in RisingWave.
+
+[AMQ Streams](https://developers.redhat.com/products/amq/overview) offers a distributed backbone that allows microservices and other applications to share data with extremely high throughput and extremely low latency. It is based on the [Strimzi](http://strimzi.io/) and [Apache Kafka](http://kafka.apache.org/) projects. AMQ Streams features a wide range of capabilities, including publish and subscribe functionality, long-term data retention, advanced queueing, replayable events, and partitioned messages for scalability.
+
+## Prerequisites
+
+Before ingesting data from RedHat AMQ Streams into RisingWave, please ensure the following:
+
+- The AMQ Streams cluster is running and accessible from your RisingWave cluster.
+- Allow the AMQ Streams cluster so that RisingWave can ingest data If authentication is required for the cluster, make sure you have the client username and password.
```
Comment (hengm3467) [inline docs/ingest/ingest-from-redhat-amq-streams.md]: Rewrite this sentence to be clearer.

File: versioned_docs/version-1.8/ingest/ingest-from-redhat-amq-streams.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
+---
+ id: ingest-from-redhat-amq-streams
```
Comment (hengm3467) [inline versioned_docs/version-1.8/ingest/ingest-from-redhat-amq-streams.md:2]: For whatever changes you made to 'docs/ingest/ingest-from-redhat...md' file, please replicate the same change in this folder 'versioned_docs/version-1.8/ingest/ingest-from-redhat-...' Thanks. @fahadullahshah261

**Comment by fahadullahshah261** [comment]
Hi, I updated MQTT PR based on Tao and Heng's suggestions. Please review it. Thanks.

File: docs/ingest/ingest-from-redhat-amq-streams.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+You can ingest data from RedHat AMQ Streams into RisingWave by using the Kafka source connector in RisingWave.
+
+[AMQ Streams](https://developers.redhat.com/products/amq/overview) offers a distributed backbone that allows microservices and other applications to share data with extremely high throughput and extremely low latency. It is based on the [Strimzi](http://strimzi.io/) and [Apache Kafka](http://kafka.apache.org/) projects. AMQ Streams features a wide range of capabilities, including publish and subscribe functionality, long-term data retention, advanced queueing, replayable events, and partitioned messages for scalability.
+
+## Prerequisites
+
+Before ingesting data from RedHat AMQ Streams into RisingWave, please ensure the following:
+
+- The AMQ Streams cluster is running and accessible from your RisingWave cluster.
+- Allow the AMQ Streams cluster so that RisingWave can ingest data. Ensure you possess the client username and password if authentication is necessary for the cluster.
```
Comment (hengm3467) [inline docs/ingest/ingest-from-redhat-amq-streams.md]: I feel this item isn't very clear. The first sentence says the cluster is accessible from your RisingWave cluster, which is the item above. Therefore, I suggest using the same wording as the MQTT one.
```suggestion
-  If authentication is required for the AMQ Streams cluster, ensure you have the client username and password.
```

File: docs/ingest/ingest-from-mqtt.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+You can ingest data from a MQTT broker into RisingWave by using the MQTT source connector in RisingWave.
+
+The [Message Queuing Telemetry Transport](https://mqtt.org/) (MQTT) protocol is ideal for connecting remote devices with a small code footprint and minimal network bandwidth. MQTT today is used in a wide variety of industries, such as automotive, manufacturing, telecommunications, oil and gas, etc.
+
+## Prerequisites
+
+Before ingesting data from MQTT into RisingWave, please ensure the following:
+
+- The MQTT server is running and accessible from your RisingWave cluster.
+- If authentication is required for the MQTT, make sure you have the client username and password. The client user must have the `subscribe` permission for the topic.
```
Comment (hengm3467) [inline docs/ingest/ingest-from-mqtt.md]:
```suggestion
- If authentication is required for the MQTT broker, make sure you have the client username and password. The client user must have the `subscribe` permission for the topic.
```

**Review by hengm3467 — APPROVED** [review]

**Comment by hengm3467** [comment]
@fahadullahshah261 You may merge this PR now.
